shared/src/main/scala/mlscript/ucs/Desugarer.scala

```diff
@@ -77,16 +78,18 @@ class Desugarer extends TypeDefs { self: Typer =>
     * this parameter indicates the location of the match root
     */
   def makeScrutinee(term: Term)(implicit matchRootLoc: Opt[Loc]): Scrutinee =
-    traceUCS(s"Making a scrutinee for $term") {
+    traceUCS(s"Making a scrutinee for `$term`") {
       term match {
         case _: Var | _: Lit => Scrutinee(N, term)
         case _ =>
-          Scrutinee(
-            S(localizedScrutineeMap.getOrElseUpdate((term.toLoc, term), {
-              Var(freshName).withLoc(term.toLoc)
-            })),
-            term
-          )
+          val localName = if (localizedScrutineeMap.containsKey(term)) {
+            localizedScrutineeMap.get(term)
+          } else {
+            val v = Var(freshName).withLoc(term.toLoc)
```

Comment:
Shouldn't this use `desugaredFrom`? In fact, maybe wee should _always_ use `desugaredFrom` instead of `withLocOf(term)` and we should make the latter also update the loc. (`withLocOf(term)` is currently like `withLocOf(term.toLoc)`).

Reply:
I didn't get "we should make the latter also update the loc". Doesn't `withLocOf(term)` update the loc?

Reply:
Oh yeah I meant that we should make the _former_ update the loc, which is what you did.